src/number_decoder.rs

```diff
@@ -252,6 +252,11 @@ static POW_10: [u64; 18] = [
     10u64.pow(17),
 ];
 
+#[cfg(target_arch = "aarch64")]
+static ONGOING_CHUNK_SIZE: u64 = POW_10[16];
+#[cfg(not(target_arch = "aarch64"))]
+static ONGOING_CHUNK_SIZE: u64 = POW_10[17];
```

Comment:
Can you explain this difference? It seems very magical to me.

Reply:
With some we do a register of 16 digits, in each chunk.

Otherwise we should do 18 digits, the max that fits under u64 max in each chunk.

I'm not clear on the n+1 issue here, but I'm surprised they're different by 1, not 2, I'll look.